src/pages/api/v1/prep-yatra/prep-log/index.ts

```diff
@@ -62,6 +64,36 @@ const handleAddLog = async (req: NextApiRequest, res: NextApiResponse) => {
       );
     }
 
+    // Trigger gamification for preplog creation
+    try {
+      let { data: gamificationData, error: gamificationError } = await updateUserPointsInDB(userId, 'PREPLOG_CREATED');
+      
+      // If user not found, create gamification record first
+      if (gamificationError && gamificationError === 'User not found') {
+        console.log('Creating gamification record for user:', userId);
+        const { data: newGamificationData, error: createError } = await addGamificationDocInDB(userId);
+        
+        if (createError) {
```

Comment:
We don't need this. We've one API which we can call as Function. We can directly pass EVENT name and kit automatically can do the rest. 

Check existing code.